esmythe_iris_BDA696.py

```diff
@@ -0,0 +1,179 @@
+"""
+Created by Ed Smythe Sept 2020
+BDA 696 @ SDSU
+
+Homework - Assignment 1 - Due Sept 8
+
+    Classic Fisher's Iris dataset (source)
+        Data, Description
+    Load the above data into a Pandas DataFrame
+    Get some simple summary statistics (mean, min, max, quartiles) using numpy
+    Plot the different classes against one another (try 5 different plots)
+        Scatter, Violin, ...
+        Try to see visual differences between them
+    Analyze and build models - Use scikit-learn
+        Use the normalizer transformer
+        Fit the transformed data against random forest classifier
+        (try other classifiers)
+        Wrap the steps into a pipeline
+        Don't worry about train/test split. (Our model will be cheating)
+    Special Notes
+        Code should just work. Make sure to lint it! No Jupyter Notebooks!
+        Work outside of your master branch and make a PR
+        Buddy should review PR, after reviewed, send it to me
+"""
+# script works for any "csv" that:
+# has 5 columns related data points gouped into 2:2:1
+# eg. IRIS data has 2 Sepal, 2 Petal and 1 class
+
+# **** search 'chg' for locations to change names for your information ****#
+
+import importlib
+import numpy as np
+import pandas as pd
+import pip
+import plotly.express as px
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import OneHotEncoder
+
+# Libraries needed
+modules = ["pandas", "numpy", "plotly", "sklearn"]
+# taken from https://bit.ly/3bCtuhO
+
+for modname in modules:

```

Comment:
While this is cool, it's not a good idea to do stuff this way. You should have done everything via requirements. We may not be using it now, but when we eventually start packaging our stuff to run on a server, we don't want to install everything at run-time when the program starts. I took away less points than normal for not having requirements since you did have this in there.

Reply:
OK, I think I do need to review that part of the class lecture and notes. I am interpreting the requirements as a responsibility of the user to make sure they have them, and not as an automated check. My intention here was for the user who ignored the readme or requirements to sill be able to use the program.